emails/ses_firebase.py

```diff
@@ -1,169 +1,144 @@
 
-    convert = {'clinical':['ClinicalTrial','GARD','GardId'], 'pubmed':['Article','GARD','GardId'], 'grant':['Project','GARD','GardId']}
-    connect_to_gard = {'clinical':'--(:Condition)--(:Annotation)--','pubmed':'--','grant':'--'}
 
-    query = 'MATCH (x:{node}){connection}(y:{gardnode}) WHERE x.DateCreatedRDAS IN {date_list} AND y.{property} IN {list} RETURN COUNT(x)'.format(node=convert[type][0], gardnode=convert[type][1], property=convert[type][2], list=list(gards.keys()), date_list=date_list, connection=connect_to_gard[type])
+def render_template(filename, data={}):
+    env = Environment(loader=FileSystemLoader(f'{sysvars.base_path}emails/'))
+    template = env.get_template(filename)
+    rendered_content = template.render(data=data)
+    return rendered_content
 
-    response = db.run(query)
-    return_data['total'] = response.data()[0]['COUNT(x)']
+def send_mail( data):
 
-    for gard in gards.keys():
-        response = db.run('MATCH (x:{node}){connection}(y:{gardnode}) WHERE x.DateCreatedRDAS IN {date_list} AND y.{property} = \"{gard}\" RETURN COUNT(x)'.format(node=convert[type][0], gardnode=convert[type][1], property=convert[type][2], gard=gard, date_list=date_list, connection=connect_to_gard[type]))
-        return_data[gard] = {'name':gards[gard],'num':response.data()[0]['COUNT(x)']}
+    # Add tabs and type to the data dictionary
+    # if data['total'] > 0 and data['email'] == '' or data['email'] == '':# for testing
+   
+    if data['total'] > 0:
+        data['email'] = '' # TEST EMAIL 
+        html_content = render_template('email_template3.html', data=data)
+        alert.send_email(f'RDAS-Alert: Rare Disease Updates Regarding Your Subscriptions',  html_content, data['email'])# change to your alert.py sending email method.you may need to adjust your method abit to read in these parameters.
+        print("finish sending enail")
 
-    return_data['update_date_end'] = date_end_string
-    return_data['update_date_start'] = date_start_string
+def get_stats(type, gard, date_start,date_end):
+    # print("master get_stats----start::",date_start, "end::",date_end)
+    db = AlertCypher(type)
+    return_data = dict()
+    # date_start_string = date_start
+    # date_end_string = date_end
+    date_start_obj = datetime.strptime(date_start, '%m/%d/%y')
+    date_end_obj = datetime.strptime(date_end, '%m/%d/%y')
 
-    return return_data
+    date_list = pd.date_range(date_start_obj, date_end_obj, freq='D').strftime('%m/%d/%y').to_list()
+    # print("date_list::",date_list)
 
-def trigger_email(type,date_start=None,date_end=None):
-    convert = {'clinical':'trials', 'pubmed':'articles', 'grant':'grants'}
-    user_data = dict()
-    cred = credentials.Certificate(sysvars.firebase_key_path)
-    firebase_admin.initialize_app(cred)
-    firestore_db = firestore.client()
+    convert = {prefix+'ctkg':['ClinicalTrial','GARD','GardId'], prefix+'pakg':['Article','GARD','GardId'], prefix+'gfkg':['Project','GARD','GardId']}
```

Comment:
manually doing "prefix+'ctkg'" seems unnecessary as in sysvars.py we already build out the DB name we are using. Example: the variable ct_db in sysvars.py